security.md

```diff
@@ -1,60 +1,190 @@
+"Private" information that does not constitute a secret but may make
+an exploit easier to achieve should be kept in private repositories.
+
+Also see the guidance on 
+[github repository contents](https://github.com/guardian/recommendations/blob/main/github.md#repository-contents)
+
+
 
 ## AWS
 
+Centralised Reporting from AWS services and guidance is available in
+[Security HQ](https://security-hq.gutools.co.uk).
+
+### Infrastructure provisioning
+
+**Infrastructure should be provisioned via cloudformation using the 
+[Guardian CDK library](https://github.com/guardian/cdk).** Many of the 
+following recommendations are encoded as defaults in the library, making
+their adoption simple.
+
+### Security patches for Compute services
+
+See the [patching guidance below](#patching)
+
+### Credentials
+
 **Do not use permanent AWS credentials.** Applications running in AWS
 should use `assumeRole` to gain access to other AWS resources. For
 local development you should use AWS profiles to manage credentials
 and these should be temporary credentials (e.g. provided by Janus).
 
+**Permanent credentials should be rotated every 365 days.** (When they
+cannot be replaced by temporary credentials.)
+
 **Ensure that at least 2 people have IAM logins for each AWS account.**  
 In the event of a VPN or Janus outage this will ensure that teams can
-still access their account. IAM users should have 2FA enabled and no
+still access their account. These IAM users should have 2FA enabled and no
 active access keys.
 
-**Use Security HQ to get an overview of our AWS accounts.**
-Centralised Reporting from AWS services, including Trusted Advisor, is
-available in [Security HQ](https://security-hq.gutools.co.uk) and can be
-used to monitor the security status of our AWS accounts.
+### SSH Access
+
+**Direct access to production infrastructure should be a last resort.** 
+Prefer automating common actions and shipping telemetry data extrernally,
+which should reduce the need.
 
 **Use SSM-Scala for SSH access.**
 [SSM-scala](https://github.com/guardian/ssm-scala#ssm-scala) is a tool
 executing commands on EC2 instances authenticated by IAM credentials.
 The SSM Agent is baked into many of our AMI images and should be
 preferentially used to permanent keys.
 
+### VPC / Security Groups
+
+**EC2 instances should be run in the private subnet of a VPC.** Only the
+load balancer of an application should reside in the public subnet. For 
+more information on VPCs and AWS see: 
+https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html.
+
+**Security Groups egress/ingress rules should be locked down as much as
+possible.** There's no reason to allow open access to any EC2 instances; 
+they should ideally be locked down to the load balancer and nothing 
+else. 
+
+**Port 22 access should be removed everywhere.** This is no longer required
+for SSH access via SSM-scala (see above.)
+
+### S3 Buckets
```

Comment:
Do you think it's worth mentioning that we should avoid (or be very careful?) when deleting s3 buckets due to risk of a 3rd party taking over the name?

Reply:
👍  done!